core/src/test/java/org/elasticsearch/action/get/GetResponseTests.java

```diff
@@ -0,0 +1,357 @@
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.get.GetField;
+import org.elasticsearch.index.get.GetResult;
+import org.elasticsearch.index.mapper.ParentFieldMapper;
+import org.elasticsearch.index.mapper.RoutingFieldMapper;
+import org.elasticsearch.index.mapper.UidFieldMapper;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.common.xcontent.XContentBuilder.DEFAULT_DATE_PRINTER;
+import static org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode;
+
+public class GetResponseTests extends ESTestCase {
+
+    public void testGetFieldToAndFromXContent() throws Exception {
+        Tuple<GetField, GetField> tuple = randomGetField();
+        GetField getField = tuple.v1();
+        GetField expectedGetField = tuple.v2();
+        XContentType xContentType = randomFrom(XContentType.values());
+        BytesReference originalBytes = toXContent(getField, xContentType, true);
+        //test that we can parse what we print out
+        GetField parsedGetField;
+        try (XContentParser parser = xContentType.xContent().createParser(originalBytes)) {
+            //we need to move to the next token, the start object one that we manually added is not expected
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+            parsedGetField = GetField.fromXContent(parser);
+            assertEquals(XContentParser.Token.END_ARRAY, parser.currentToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+        }
+        assertEquals(expectedGetField, parsedGetField);
+        //print the parsed object out and test that the output is the same as the original output
+        BytesReference finalBytes = toXContent(parsedGetField, xContentType, true);
+        assertEquals(originalBytes, finalBytes);
+    }
+
+    public void testGetFieldToXContent() throws IOException {
+        GetField getField = new GetField("field", Arrays.asList("value1", "value2"));
+        String output = toXContent(getField, XContentType.JSON, true).utf8ToString();
+        assertEquals("{\"field\":[\"value1\",\"value2\"]}", output);
+    }
+
+    public void testGetFieldEqualsAndHashcode() {
+        checkEqualsAndHashCode(randomGetField().v1(), GetResponseTests::copyGetField, GetResponseTests::mutateGetField);
+    }
+
+    public void testGetResultToAndFromXContent() throws Exception {
```

Comment:
At this point I would try to avoid too much hierarchy in the tests. In the query refactoring we ended up centralizing a lot of helpers in `AbstractQueryTestCase` but this ended up as a huge (>1000 loc) class and created a bunch of dependencies between the individual tests. To reuse freuqently needed helper methods I'd rather use utility classes with static methods where possible instead of creating abstract base classes too early. That said, we might need to revisit that decision at a later point.